### DISCUSSION TITLE
Getting response and playing video after?

### DISCUSSION THREAD

**Discussion by kasmith** [jspsych/jsPsych#3012] [Q&A]
I'm trying to create a stimulus for which people will see the first frame of a movie, have to press a button on the keyboard to register a prediction of what will happen, and finally the video plays to give them feedback.

Currently I have this set up as two separate VideoKeyboardResponse trials with the same stimuli -- one that disables autoplay and sets response_ends_trial to true, and the second that sets NO_KEYS with autoplay and trial_ends_after_video. However, this leads to a flash between the response and video playing, as the video needs to be re-added to the screen.

So I was wondering if there was a way to do this in a single trial object (without cracking into the plugin itself)?

Thanks!

**Comment by ChristopheBossens** [comment]
Not sure if this is possible without changing the plugin code, because as you mention, with a different stimulus it needs to be re-added to the screen.

I found out that adjusting the plugin code is not that difficult, you can just add the line 

`video_element.play();`

after the `after_response()` function (line 257 in the most recent version of jsPsych). If you use the following code to create the trial, it starts to play after the first response:

```
    var trial = {
        type: jsPsychVideoKeyboardResponse,
        stimulus: [
            'movie.webm'
        ],
        response_ends_trial: false,
        trial_ends_after_video: true,
        autoplay: false
        };
```
